tlsutil/config_test.go

```diff
@@ -538,20 +538,60 @@ func TestConfigurator_CommonTLSConfigGetClientCertificate(t *testing.T) {
 	c, err := NewConfigurator(Config{}, nil)
 	require.NoError(t, err)
 
-	cert, err := c.commonTLSConfig(false).GetCertificate(nil)
+	cert, err := c.commonTLSConfig(false).GetClientCertificate(nil)
 	require.NoError(t, err)
 	require.Nil(t, cert.Certificate)
 
 	c.manual.cert = &tls.Certificate{}
-	cert, err = c.commonTLSConfig(false).GetCertificate(nil)
+	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
+	require.NoError(t, err)
+	require.Equal(t, c.manual.cert, cert)
+
+	c.manual.cert = &tls.Certificate{Certificate: [][]byte{}}
+	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
 	require.NoError(t, err)
 	require.Equal(t, c.manual.cert, cert)
 
+	c.autoEncrypt.cert = &tls.Certificate{Certificate: [][]byte{}}
+	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
+	require.NoError(t, err)
+	require.Equal(t, c.autoEncrypt.cert, cert)
+
+	c.autoEncrypt.cert = &tls.Certificate{}
 	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
 	require.NoError(t, err)
 	require.Equal(t, c.manual.cert, cert)
 }
 
+func TestConfigurator_CommonTLSConfigGetCertificate(t *testing.T) {
+	c, err := NewConfigurator(Config{}, nil)
+	require.NoError(t, err)
+
+	cert, err := c.commonTLSConfig(false).GetClientCertificate(nil)
+	require.NoError(t, err)
+	require.Nil(t, cert.Certificate)
+
+	c.autoEncrypt.cert = &tls.Certificate{}
+	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
+	require.NoError(t, err)
+	require.Equal(t, c.autoEncrypt.cert, cert)
+
+	c.autoEncrypt.cert = &tls.Certificate{Certificate: [][]byte{}}
+	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
+	require.NoError(t, err)
+	require.Equal(t, c.autoEncrypt.cert, cert)
+
+	c.manual.cert = &tls.Certificate{Certificate: [][]byte{}}
```

Comment:
Should we add some actual payload to this Certificate that means it can't accidentally compare equal to the auto one above due to deep-equals rather than exact pointer checking?

I'm actually not sure this assertion prooves anything currently because I think it would still pass even if we were returning the auto cert not the manual cert after this since they both have equivalent values and `require.Equal` does a deep equality not a pointer match IIRC.

Making them obviously different would make the intent of this test clearer too.